soc/nordic/common/pm_s2ram.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/linker/sections.h>
+#include <zephyr/arch/common/pm_s2ram.h>
+#include <hal/nrf_resetinfo.h>
+#include <pm_s2ram.h>
+
+
+#define NVIC_MEMBER_SIZE(member) ARRAY_SIZE(((NVIC_Type *)0)->member)
```

Comment:
please, explicitly include CMSIS header (where these types reside). Also, header for ARRAY_SIZE is missing.

Reply:
done

Reply:
headers should be sorted properly

Reply:
also, files are still in common folder